### DISCUSSION TITLE
OpenMCT gives problems when added as dependency

### DISCUSSION THREAD

**Discussion by gallacci97** [nasa/openmct#4185] [Community Q&A] [answered]
Hi all, 
I'm trying to create a new node project based on OpenMCT. When I type the command to add it as dependency it gives me back al lot of errors referring to modules which are deprecated or not used anymore as you can see below. Did you encountered the same problem? Do you know how can I solve it?
My guess is that my npm and/or node versions are too new and maybe I have to use older versions but I prefer not to.

C:\Users\s\source\js_repos\new_method\from_scratch>npm install github:nasa/openmct
npm ERR! code 1
npm ERR! git dep preparation failed
npm ERR! command C:\Program Files (x86)\nodejs\node.exe C:\Users\s\AppData\Roaming\npm\node_modules\npm\bin\npm-cli.js install --force --cache=C:\Users\s\AppData\Local\npm-cache --prefer-offline=false --prefer-online=false --offline=false --no-progress --no-save --no-audit --include=dev --include=peer --include=optional --no-package-lock-only --no-dry-run
npm ERR! npm WARN using --force Recommended protections disabled.
npm ERR! npm WARN deprecated gulp-header@1.8.12: Removed event-stream from gulp-header
npm ERR! npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm ERR! npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm ERR! npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm ERR! npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm ERR! npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm ERR! npm WARN deprecated babel-eslint@10.0.3: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm ERR! npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm ERR! npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm ERR! npm WARN deprecated coffee-script@1.12.7: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)
npm ERR! npm WARN deprecated tar@2.2.2: This version of tar is no longer supported, and will not receive security updates. Please upgrade asap.
npm ERR! npm WARN deprecated core-js@2.6.12: core-js@<3.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
npm ERR! npm ERR! code 1
npm ERR! npm ERR! path C:\Users\s\AppData\Local\npm-cache\_cacache\tmp\git-cloneuvDHHn\node_modules\node-sassnpm ERR! npm ERR! command failed
npm ERR! npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c node scripts/build.js
npm ERR! npm ERR! Building: C:\Program Files (x86)\nodejs\node.exe C:\Users\s\AppData\Local\npm-cache\_cacache\tmp\git-cloneuvDHHn\node_modules\node-gyp\bin\node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library=
npm ERR! npm ERR! gyp info it worked if it ends with ok
npm ERR! npm ERR! gyp verb cli [
npm ERR! npm ERR! gyp verb cli   'C:\\Program Files (x86)\\nodejs\\node.exe',
npm ERR! npm ERR! gyp verb cli   'C:\\Users\\s\\AppData\\Local\\npm-cache\\_cacache\\tmp\\git-cloneuvDHHn\\node_modules\\node-gyp\\bin\\node-gyp.js',
npm ERR! npm ERR! gyp verb cli   'rebuild',
npm ERR! npm ERR! gyp verb cli   '--verbose',
npm ERR! npm ERR! gyp verb cli   '--libsass_ext=',
npm ERR! npm ERR! gyp verb cli   '--libsass_cflags=',
npm ERR! npm ERR! gyp verb cli   '--libsass_ldflags=',
npm ERR! npm ERR! gyp verb cli   '--libsass_library='
npm ERR! npm ERR! gyp verb cli ]
npm ERR! npm ERR! gyp info using node-gyp@3.8.0
npm ERR! npm ERR! gyp info using node@14.17.0 | win32 | ia32
npm ERR! npm ERR! gyp verb command rebuild []
npm ERR! npm ERR! gyp verb command clean []
npm ERR! npm ERR! gyp verb clean removing "build" directory
npm ERR! npm ERR! gyp verb command configure []
npm ERR! npm ERR! gyp verb check python checking for Python executable "python2" in the PATH
npm ERR! npm ERR! gyp verb `which` failed Error: not found: python2
npm ERR! npm ERR! gyp verb `which` failed     at getNotFoundError (C:\Users\s\AppData\Local\npm-cache\_cacache\tmp\git-cloneuvDHHn\node_modules\node-gyp\node_modules\which\which.js:13:12)
npm ERR! npm ERR! gyp verb `which` failed     at F (C:\Users\s\AppData\Local\npm-cache\_cacache\tmp\git-cloneuvDHHn\node_modules\node-gyp\node_modules\which\which.js:68:19)
npm ERR! npm ERR! gyp verb `which` failed     at E (C:\Users\s\AppData\Local\npm-cache\_cacache\tmp\git-cloneuvDHHn\node_modules\node-gyp\node_modules\which\which.js:80:29)
npm ERR! npm ERR! gyp verb `which` failed     at C:\Users\s\AppData\Local\npm-cache\_cacache\tmp\git-cloneuvDHHn\node_modules\node-gyp\node_modules\which\which.js:89:16
npm ERR! npm ERR! gyp verb `which` failed     at C:\Users\s\AppData\Local\npm-cache\_cacache\tmp\git-cloneuvDHHn\node_modules\isexe\index.js:42:5
npm ERR! npm ERR! gyp verb `which` failed     at C:\Users\s\AppData\Local\npm-cache\_cacache\tmp\git-cloneuvDHHn\node_modules\isexe\windows.js:36:5
npm ERR! npm ERR! gyp verb `which` failed     at FSReqCallback.oncomplete (fs.js:192:21)
npm ERR! npm ERR! gyp verb `which` failed  python2 Error: not found: python2
npm ERR! npm ERR! gyp verb `which` failed     at getNotFoundError (C:\Users\s\AppData\Local\npm-cache\_cacache\tmp\git-cloneuvDHHn\node_modules\node-gyp\node_modules\which\which.js:13:12)
npm ERR! npm ERR! gyp verb `which` failed     at F (C:\Users\s\AppData\Local\npm-cache\_cacache\tmp\git-cloneuvDHHn\node_modules\node-gyp\node_modules\which\which.js:68:19)
npm ERR! npm ERR! gyp verb `which` failed     at E (C:\Users\s\AppData\Local\npm-cache\_cacache\tmp\git-cloneuvDHHn\node_modules\node-gyp\node_modules\which\which.js:80:29)
npm ERR! npm ERR! gyp verb `which` failed     at C:\Users\s\AppData\Local\npm-cache\_cacache\tmp\git-cloneuvDHHn\node_modules\node-gyp\node_modules\which\which.js:89:16
npm ERR! npm ERR! gyp verb `which` failed     at C:\Users\s\AppData\Local\npm-cache\_cacache\tmp\git-cloneuvDHHn\node_modules\isexe\index.js:42:5
npm ERR! npm ERR! gyp verb `which` failed     at C:\Users\s\AppData\Local\npm-cache\_cacache\tmp\git-cloneuvDHHn\node_modules\isexe\windows.js:36:5
npm ERR! npm ERR! gyp verb `which` failed     at FSReqCallback.oncomplete (fs.js:192:21) {
npm ERR! npm ERR! gyp verb `which` failed   code: 'ENOENT'
npm ERR! npm ERR! gyp verb `which` failed }
npm ERR! npm ERR! gyp verb check python checking for Python executable "python" in the PATH
npm ERR! npm ERR! gyp verb `which` succeeded python C:\Users\s\AppData\Local\Programs\Python\Python39\python.EXE
npm ERR! npm ERR! gyp ERR! configure error
npm ERR! npm ERR! gyp ERR! stack Error: Command failed: C:\Users\s\AppData\Local\Programs\Python\Python39\python.EXE -c import sys; print "%s.%s.%s" % sys.version_info[:3];
npm ERR! npm ERR! gyp ERR! stack   File "<string>", line 1
npm ERR! npm ERR! gyp ERR! stack     import sys; print "%s.%s.%s" % sys.version_info[:3];
npm ERR! npm ERR! gyp ERR! stack                       ^
npm ERR! npm ERR! gyp ERR! stack SyntaxError: invalid syntax
npm ERR! npm ERR! gyp ERR! stack
npm ERR! npm ERR! gyp ERR! stack     at ChildProcess.exithandler (child_process.js:319:12)
npm ERR! npm ERR! gyp ERR! stack     at ChildProcess.emit (events.js:376:20)
npm ERR! npm ERR! gyp ERR! stack     at maybeClose (internal/child_process.js:1055:16)
npm ERR! npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
npm ERR! npm ERR! gyp ERR! System Windows_NT 10.0.19041
npm ERR! npm ERR! gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Users\\s\\AppData\\Local\\npm-cache\\_cacache\\tmp\\git-cloneuvDHHn\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
npm ERR! npm ERR! gyp ERR! cwd C:\Users\s\AppData\Local\npm-cache\_cacache\tmp\git-cloneuvDHHn\node_modules\node-sass
npm ERR! npm ERR! gyp ERR! node -v v14.17.0
npm ERR! npm ERR! gyp ERR! node-gyp -v v3.8.0
npm ERR! npm ERR! gyp ERR! not ok
npm ERR! npm ERR! Build failed with error code: 1
npm ERR!
npm ERR! npm ERR! A complete log of this run can be found in:
npm ERR! npm ERR!     C:\Users\s\AppData\Local\npm-cache\_logs\2021-09-03T15_35_14_002Z-debug.log

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\s\AppData\Local\npm-cache\_logs\2021-09-03T15_35_15_912Z-debug.log

**Comment by akhenry** [comment]
Hi, have you tried building with our pre-built npm package - https://www.npmjs.com/package/openmct ?

**Reply by gallacci97** [reply]
Thank you for your response.
During these days I tried different version of node from 8 to 16 with different openmct branches and I manage to install it on a ubuntu machine but on windows still no success (for corporate reasons I need it on win10). 
In fact I didn't know about the npm package but I still get a similar error typing "npm install openmct" or "npm i openmct". 
I attach here my last try.
[log.txt](https://github.com/nasa/openmct/files/7149429/log.txt)

**Comment by jvigliotta** [comment, marked as answer]
Hello @gallacci97 I see you were using node 14... I believe that SHOULD work, but we do know node 12 is working if you want to try that as well.

**Reply by gallacci97** [reply]
Hi @jvigliotta, sorry for the late reply. After some trial and error I managed to install it successfully with node 12. Thank you!
